### PR TITLE
benchmark-compatibility improvements

### PR DESCRIPTION
**Pull request by staceysv**
A collection of minor changes mostly to clean up the logging to W&B and simplify benchmark comparison.
Very open to feedback/improvements, especially on any conventions, naming, and any useful logs I may have removed.

* biggest change: refactor to log all PDFs to one "Media" panel section instead of a separate section for each single doc. Logging for W&B-native bounding boxes is set up but not enabled--boxes won't show up correctly until we a) resize all logged documents to the same fixed size and/or b) make changes to the W&B CLI and/or c) do some further debugging. log_pdfs_with_wandb() is basically a clone of log_pdf() with this extension--now that it's stable/if we like it, we could a) remove log_pdfs() and/or b) take out all the references to logging W&B bounding boxes until we can get those to work. The pdfplumber native boxes are pretty great, though could probably be cleaned up/thresholded/organized further. I organized the colors to signify different classes, and green to signify the ground truth.
* second biggest change: add an "avg_acc" field to use as the main ranking for the benchmark, and rename fields for the benchmark slightly ("contract_num" => "contractid", "gross_amount" => "amount") so I don't have to mess with the very narrow columns for the leaderboard/keep trying to convince our main front-end person on this point :).

minor tail
* fixes a random seed for a deterministic train/val data split (also sets this to as a TF random seed for reproducibility, no strong attachment to this)
* removes references to WANDB_ENTITY, will require all users to login with their respective wandb accounts
* sets a different default WANDB_PROJECT, "deepform_v1"

### CODE REVIEW

**Review by moredatarequired — COMMENTED** [review]
Thanks! This looks great. I made a bunch of comments, but most of them don't matter much, they're just code preferences. I think I would like the print statements turned into logging statements, but everything else is of the form 'you did this X way, I would have done it Y way' and are totally your call whether you take me up on my suggestion.

File: deepform/train.py [hunk 1]
```diff
@@ -51,9 +53,19 @@ def compute_accuracy(model, config, dataset, num_to_test, print_results, log_pat
 
         if print_results:
             print(f"file_id:{slug}")
+
+        # track all logging information for this document
+        doc_log = {}
+        for log_key in ["pred_text", "true_text", "score", "field", "match"]:
+            doc_log[log_key] = []
```
[the remark this answers is not in the document]
Comment (staceysv) [reply deepform/train.py:60]: good catch!

File: deepform/train.py [hunk 2]
```diff
@@ -89,6 +99,13 @@ def __init__(self, config, run_timestamp, dataset, logname):
         self.logname = logname
         self.log_path = LOG_DIR / "predictions" / run_timestamp
 
+    def mean_field_acc(self, acc_dict):
+        # return the average accuracy across all fields
+        # advertiser, contract_num, flight_from, flight_to, gross_amount
+        # we may want to adjust the relative weighting of these in the future
+        total_acc = sum(v for v in acc_dict.values())
+        return total_acc / 5.0
```
[the remark this answers is not in the document]
Comment (staceysv) [reply deepform/train.py:105]: ok yep looks good! i wasn't sure if there would be other metrics in there :)

File: deepform/train.py [hunk 3]
```diff
@@ -115,14 +132,32 @@ def on_epoch_end(self, epoch, logs):
         acc_str = re.sub(r"\s+", " ", acc.to_string())
 
         print(f"This epoch {self.logname}: {acc_str}")
-        wandb.log({self.logname: acc_str})
+        acc_dict = acc.to_dict()
```
[the remark this answers is not in the document]
Comment (staceysv) [reply deepform/train.py:132]: hmmm ok cool! at some point it was complaining about logging to W&B...

File: deepform/train.py [hunk 4]
```diff
@@ -115,14 +132,32 @@ def on_epoch_end(self, epoch, logs):
...
+            {
+                "amount": acc_dict["gross_amount"],
+                "flight_to": acc_dict["flight_to"],
+                "flight_from": acc_dict["flight_from"],
+                "contractid": acc_dict["contract_num"],
+                "advertiser": acc_dict["advertiser"],
+            }
+        )
+        # compute average accuracy
+        wandb.log({"avg_acc": self.mean_field_acc(acc_dict)})
```
[the remark this answers is not in the document]
Comment (staceysv) [reply deepform/train.py]: NICE.

File: deepform/pdfs.py [hunk 5]
```diff
@@ -47,6 +47,118 @@ def download_from_remote(local_path):
...
+    target_idx = [idx for (idx, val) in enumerate(doc.labels) if val == 1]
+
+    class_ids_by_field = {
+        "gross_amount": 0,
+        "flight_to": 1,
+        "flight_from": 2,
+        "contract_num": 3,
+        "advertiser": 4,
+    }
+    class_id_to_label = {int(v): k for k, v in class_ids_by_field.items()}
```
[the remark this answers is not in the document]
Comment (staceysv) [reply deepform/pdfs.py:72]: I appreciate this! I was trying to move quickly & not have wandb complain--will try to get this in.

**Comment by staceysv** [comment]
awesome, thanks so much @moredatarequired ! was moving quickly, hence the copy&pastes :), accepted all of these except the TokenType suggestions (for now!), now polishing tiny inconsistencies. Much cleaner overall :)

File: deepform/train.py [hunk 6]
```diff
@@ -115,14 +129,31 @@ def on_epoch_end(self, epoch, logs):
...
+            {
+                "amount": acc_dict["gross_amount"],
+                "flight_to": acc_dict["flight_to"],
+                "flight_from": acc_dict["flight_from"],
+                "contractid": acc_dict["contract_num"],
+                "advertiser": acc_dict["advertiser"],
+            }
+        )
+        # compute average accuracy
+        wandb.log({"avg_acc": acc_dict.mean()})
```
Comment (moredatarequired) [inline deepform/train.py:144]: Oops, this also needs to change to directly use the pd.Series `acc`.

```suggestion
        wandb.log(
            {
                "amount": acc["gross_amount"],
                "flight_to": acc["flight_to"],
                "flight_from": acc["flight_from"],
                "contractid": acc["contract_num"],
                "advertiser": acc["advertiser"],
            }
        )
        # compute average accuracy
        wandb.log({"avg_acc": acc.mean()})
```

Comment (moredatarequired) [reply deepform/train.py:144]: Hmm, actually I would probably just rename and use to_dict for the log:
```suggestion
        wandb.log(acc.rename({"gross_amount": "amount", "contract_num": "contractid"}).to_dict())
        # compute average accuracy
        wandb.log({"avg_acc": acc.mean()})
```

**Comment by moredatarequired** [comment]
@staceysv , thank you so much for all this work! I really appreciate all the usability enhancements we get this way in W&B.

File: deepform/pdfs.py [hunk 7]
```diff
@@ -47,6 +47,77 @@ def download_from_remote(local_path):
...
+            "box_data": true_bboxes,
+            "class_labels": class_id_to_label,
+        },
+    }
+    wandb.log(
+        {
+            "pdf/"
+            + fname.name
+            + ":"
+            + str(pagenum): wandb.Image(im.annotated, boxes=boxes)
```
Comment (moredatarequired) [inline deepform/pdfs.py:116]:
```suggestion
            f"pdf/{fname.name}:{pagenum}": wandb.Image(im.annotated, boxes=boxes)
```
(I just find this more readable)

**Review by moredatarequired — APPROVED** [review]

**Pull request merged**